### PR TITLE
lighttpd deploy hook

### PR DESCRIPTION
**Pull request by michaelw**
* verbatim copy from haproxy.sh, s/haproxy/lighttpd
* enable issuer

### CODE REVIEW

File: deploy/lighttpd.sh [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  # Some defaults
+  DEPLOY_LIGHTTPD_PEM_PATH_DEFAULT="/etc/lighttpd"
+  DEPLOY_LIGHTTPD_PEM_NAME_DEFAULT="${_cdomain}.pem"
+  DEPLOY_LIGHTTPD_BUNDLE_DEFAULT="no"
+  DEPLOY_LIGHTTPD_ISSUER_DEFAULT="yes"
+  DEPLOY_LIGHTTPD_RELOAD_DEFAULT="true"
+
+  if [ -f "${DOMAIN_CONF}" ]; then
+    # shellcheck disable=SC1090
+    . "${DOMAIN_CONF}"
```
Comment (Neilpang) [inline deploy/lighttpd.sh:59]: don't import DOMAIN_CONF

File: deploy/lighttpd.sh [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  fi
+
+  _debug _cdomain "${_cdomain}"
+  _debug _ckey "${_ckey}"
+  _debug _ccert "${_ccert}"
+  _debug _cca "${_cca}"
+  _debug _cfullchain "${_cfullchain}"
+
+  # PEM_PATH is optional. If not provided then assume "${DEPLOY_LIGHTTPD_PEM_PATH_DEFAULT}"
+  if [ -n "${DEPLOY_LIGHTTPD_PEM_PATH}" ]; then
```
Comment (Neilpang) [inline deploy/lighttpd.sh:69]: read these DEPLOY_*  variables  by "_getdeployconf()" function.

see example: https://github.com/acmesh-official/acme.sh/blob/master/deploy/docker.sh

**Comment by gstrauss** [comment]
@michaelw wrote:
> * verbatim copy from haproxy.sh, s/haproxy/lighttpd
> * enable issuer

@Neilpang please advise how to proceed here.  If you are asking for changes to this PR, should `deploy/haproxy.sh` also be modified the same way?

**Comment by Neilpang** [comment]
>> should deploy/haproxy.sh also be modified the same way?

Yes, but it was born far earlier than this one.   It will be changed sometime in future.

**Comment by michaelw** [comment]
Thanks!

**Comment by gstrauss** [comment]
> Thanks! Can this be closed then?

This will automatically be closed if #3687 is merged.

Now, we both wait for someone to look at #3687 ...

**Pull request closed without merging**